.github/actions/create-deployment/src/index.js

```diff
@@ -0,0 +1,176 @@
+// @ts-check
+
+import core from '@actions/core';
+import github from '@actions/github';
+
+const NAME_KEY = 'name';
+const DEPLOYMENT_ID_OUTPUT_KEY = 'deployment_id';
+const SUCCESS_STATE = 'success';
+const INACTIVE_STATE = 'inactive';
+
+/**
+ * @see https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+ */
+const REGEX = {
+    BRANCH: /refs\/heads\/(.*)/,
+    TAG: /refs\/tags\/(.*)/,
+
+    // If the above two fail to match, then use GITHUB_HEAD_REF instead of regex match.
+    PULL_REQUEST: /refs\/pull\/(\d+)\/merge/,
+};
+
+/**
+ * If triggered by PUSH or TAG, return a regex match.
+ * For pull requests, use the head ref.
+ *
+ * @returns {string} Current branch the action is running for.
+ */
+function getBranchName() {
+    const ref = process.env['GITHUB_REF'] ?? '';
+
+    const branchMatch = REGEX.BRANCH.exec(ref);
+    if (branchMatch) {
+        return branchMatch[1] ?? '';
+    }
+
+    const tagMatch = REGEX.TAG.exec(ref);
+    if (tagMatch) {
+        return tagMatch[1] ?? '';
+    }
+
+    return process.env['GITHUB_HEAD_REF'] ?? '';
+}
```

Comment:
Why is this done with regex instead of the default [Github context object](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)? If I'm not mistaken you can just use `github.ref_name` and `github.ref_type`.

Reply:
Oh yeah I should mention that. The ref for all PRs always looks like `refs/pull/<PR NUM>/merge`. If you use this as the ref for a deployment, it gives you absolutely no information on the deployment dashboard. It will always just say "merging PR# into main" and that's it. Literally GitHub doesn't use this ref either; you have to use the literal short branch name or `refs/heads/<branch name>` for the commit title to show up on the deployment info, and most importantly for the PR to receive the "deployed" buttons.

Reply:
Oh I see yea I forgot about that. We love Github actions!